options.go

```diff
@@ -25,6 +25,14 @@ func Domain(domain string) Option {
 	}
 }
 
+// Exclude paths from CSRF checks, for example endpoints that would receive
+// webhooks.
+func Exclude(paths ...string) Option {
```

Comment:
This will need to be documented much more clearly - e.g. what does it match on? Does it support wildcards? Prefixes? (paths only, no, no).

There's a risk of misconfiguration leading to a security issue for many people here, and thus the docs need to be airtight.

Reply:
PS: I should add that I've rejected this feature before, and primarily because the expectations around it are unclear, and at risk of snowballing.

My advice prior has been to use the [subrouter](https://godoc.org/github.com/gorilla/mux#Route.Subrouter) or route-groups functionality of your HTTP router of choice to split out what routes need CSRF (most; by default) and what should be excluded (some API endpoints) using the URL matching there.

Otherwise, any modestly-sized app is going to have 10-20 "exclude paths" to maintain?